sdk/test-utils/perf/src/batchPerfTest.ts

```diff
@@ -0,0 +1,200 @@
+   * Apply this method on the client to get the proxy tool support.
+   *
+   * Note: Client must expose the pipeline property which is required for the perf framework to add its policies correctly
+   */
+  public configureClient<T>(client: T & { pipeline: Pipeline }): T {
+    if (this.testProxy) {
+      this.testProxyHttpClient = new TestProxyHttpClient(
+        this.testProxy,
+        this.parsedOptions["insecure"].value ?? false
+      );
+      client.pipeline.addPolicy(
+        testProxyHttpPolicy(
+          this.testProxyHttpClient,
+          this.testProxy.startsWith("https"),
+          this.parsedOptions["insecure"].value ?? false
+        )
+      );
+    }
+    return client;
+  }
+
+  /**
+   * Runs the test in scope repeatedly, without waiting for any promises to finish,
+   * as many times as possible until durationMilliseconds is reached.
+   * For each test run, it will report one more completedOperations on the PerfParallel given,
+   * as well as the lastMillisecondsElapsed that reports the last test execution's elapsed time in comparison
+   * to the beginning of the execution of runLoop.
+   *
+   * @param parallel Object where to log the results from each execution.
+   * @param durationMilliseconds When to abort any execution.
+   * @param abortController Allows us to send through a signal determining when to abort any execution.
+   */
+  public async runAll(
+    parallel: PerfParallel,
+    durationMilliseconds: number,
+    abortController: AbortController
+  ): Promise<void> {
+    parallel.completedOperations = 0;
+    parallel.lastMillisecondsElapsed = 0;
+    const start = process.hrtime();
+    while (!abortController.signal.aborted) {
+      const completedOperations = await this.runBatch(abortController.signal);
+
+      const elapsed = process.hrtime(start);
+      const elapsedMilliseconds = elapsed[0] * 1000 + elapsed[1] / 1000000;
+
+      parallel.completedOperations += completedOperations;
+      parallel.lastMillisecondsElapsed = elapsedMilliseconds;
+
+      // In runTest we create a setTimeout that is intended to abort the abortSignal
+      // once the durationMilliseconds have elapsed. That setTimeout might not get queued
+      // on time through the event loop, depending on the number of operations we might be executing.
+      // For this reason, we're also manually checking the elapsed time here.
+      if (abortController.signal.aborted || elapsedMilliseconds > durationMilliseconds) {
+        abortController.abort();
+        break;
+      }
+    }
+  }
+
+  /**
+   * postSetup
+   */
+  public async postSetup() {
+    // Records requests(in the run method) for all the instantiated PerfTest classes,
+    // and asks the proxy-tool to start playing back for future requests.
+    //
+    // Only be executed when the test-proxies option is provided.
+    // Expects the proxy tool to be up and running
+    if (this.testProxy) return this.recordAndStartPlayback();
```

Comment:
```suggestion
  /**
   * `postSetup` records requests (in the run method) for all the instantiated `PerfTest` classes,
   * and asks the proxy-tool to start playing back for future requests.
   *
   * Should be executed only when the `test-proxies` option is provided.
   * Expects the proxy tool to be up and running.
   */
  public async postSetup() {
    if (this.testProxy) return this.recordAndStartPlayback();
```

This comment makes more sense to me in the descriptor of the method, rather than inside the method. The descriptor is used by the editors (I believe), so having only `postSetup` doesn’t do much.

Reply:
The thing is.. it is subject to change, we may add new things in the `postSetup` and not just the recorder.
I'll instead add when it runs, what it can do.